src/rust/engine/sharded_lmdb/src/lib.rs

```diff
@@ -159,7 +156,7 @@ impl ShardedLmdb {
       .set_flags(EnvironmentFlags::NO_SYNC | EnvironmentFlags::NO_TLS)
       // 2 DBs; one for file contents, one for leases.
       .set_max_dbs(2)
-      .set_map_size(max_size_bytes / page_size::get())
+      .set_map_size(max_size)
```

Comment:
My read of the docs for this method is that it is supposed to be bytes, but that it should be a multiple of the page size.